ui/src/session.ts

```diff
@@ -127,11 +126,20 @@ export const createKeystore = (passphrase: string): Promise<null> => {
   return api.set<unknown>(`keystore`, { passphrase });
 };
 
-const updateSettings = (settings: Settings): Promise<void> =>
-  api
-    .set<Settings>(`session/settings`, settings)
-    .then(fetchSession)
-    .catch((err: error.Error) => notification.error(err.message));
+const updateSettings = async (settings: Settings): Promise<void> => {
+  try {
+    api.set<Settings>(`session/settings`, settings);
```

Comment:
Don't we need to `await` here as well? For the 2 api calls happen serially.

Reply:
Yeah, we do. Fixed in https://github.com/radicle-dev/radicle-upstream/pull/1251/commits/98f5f6067046f1f39ed9cd6e4e013958476e691f